.github/workflows/cicd-2-publish.yaml

```diff
@@ -50,13 +55,77 @@ jobs:
     steps:
       - name: "Checkout code"
         uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: "Generate GitHub App token"
+        uses: actions/create-github-app-token@v1
+        id: app-token
+        with:
+          app-id: ${{ secrets.GITHUB_APP_ID }}
+          private-key: ${{ secrets.GITHUB_APP_PRIVATE_KEY }}
+      - name: "Import GPG key and configure signing"
+        if: ${{ secrets.GITHUB_APP_SIGNING_KEY != '' }}
+        env:
+          GPG_SIGNING_KEY: ${{ secrets.GITHUB_APP_SIGNING_KEY }}
+          GPG_KEY_ID: ${{ secrets.GITHUB_APP_SIGNING_KEY_ID }}
+          GITHUB_APP_SIGNING_KEY_PASSPHRASE: ${{ secrets.GITHUB_APP_SIGNING_KEY_PASSPHRASE }}
+        run: |
```

Comment:
Another thing to consider after merge - this section could be a good candidate to extract into a self contained script (or composable action) for better testability if we want to promote it as a recommended practice for others to follow. We could assume that once it works, it works, but occasionally the environment may change and require an update. Having it isolated would make it easier to pinpoint and resolve such issues. It would also help with reuse if we later decide to standardise this signing setup across multiple repositories.